bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/TypeConverter.java

```diff
@@ -365,6 +365,22 @@ public static Set<Type> getConvertibleTypes(Object inputValue, Type targetType,
         return convertibleTypes;
     }
 
+    private static boolean isDirectlyConvertible(Object inputValue, Type inputValueType, Type memType,
```

Comment:
I think we can refactor the same logic not to have duplicate function calls, like done in https://github.com/HindujaB/ballerina-lang/pull/2/files. There I have also added tests with tuples having rest types.